### PR TITLE
Use one filter if same

### PR DESCRIPTION
**Pull request by schaumb**

### CODE REVIEW

File: src/chart/animator/animation.cpp [hunk 1]
```diff
@@ -65,14 +65,22 @@ void Animation::addKeyframe(const Gen::PlotPtr &next,
...
 	else if (strategy == RegroupStrategy::aggregate) {
-		auto &&andFilter = Data::Filter(
-		    [srcFilter = target->getOptions()->dataFilter,
-		        trgFilter = next->getOptions()->dataFilter](
-		        const Data::RowWrapper &row)
-		    {
-			    return srcFilter.match(row) && trgFilter.match(row);
-		    },
-		    0);
+		auto &srcFilter = target->getOptions()->dataFilter;
```
Comment (simzer) [inline src/chart/animator/animation.cpp:68]: could we put this into Data::Filter::operator&&()

**Comment by schaumb** [comment]
Review: The condition is not the final. We need to keep the dimensions until labellevel if only dimensions on the stackaxis.

File: src/chart/animator/animation.cpp [hunk 2]
```diff
@@ -133,7 +127,7 @@ void Animation::addKeyframe(const Gen::PlotPtr &next,
 	addKeyframe(begin,
 	    next,
 	    options,
-	    strategy == RegroupStrategy::drilldown);
+	    strategy != RegroupStrategy::fade);
```
Comment (simzer) [inline src/chart/animator/animation.cpp:130]: I suspect that this change should be done for the first addKeyframe in line 117

**Review by simzer — APPROVED** [review]

File: src/chart/animator/animation.cpp (hunk 2 above)
Comment (schaumb) [reply src/chart/animator/animation.cpp:130]: OK I'll do that in the next PR

**Pull request merged**